pkg/jobconfig/files.go

```diff
@@ -242,16 +248,60 @@ func WriteToDir(jobDir, org, repo string, jobConfig *prowconfig.JobConfig) error
 	return nil
 }
 
+func labelGeneratedJobs(jobConfig *prowconfig.JobConfig, label string) {
+	for _, jobs := range jobConfig.Presubmits {
+		for _, job := range jobs {
+			if _, isGenerated := job.Labels[ProwJobLabelGenerated]; isGenerated {
+				job.Labels[ProwJobLabelGenerated] = label
+			}
+		}
+	}
+	for _, jobs := range jobConfig.Postsubmits {
+		for _, job := range jobs {
+			if _, isGenerated := job.Labels[ProwJobLabelGenerated]; isGenerated {
+				job.Labels[ProwJobLabelGenerated] = label
+			}
+		}
+	}
+}
+
+func pruneStaleGeneratedJobs(jobConfig *prowconfig.JobConfig, staleLabel string) {
```

Comment:
I am not saying that this is wrong, but I personally try to avoid re-arranging the slice with the indexing. I don't trust it :)
How about:
```suggestion
func pruneStaleGeneratedJobs(jobConfig *prowconfig.JobConfig, staleLabel string) {
	for repo, jobs := range jobConfig.Presubmits {
		tempJobs := []prowconfig.Presubmit{}
		for _, job := range jobs {
			if label, isGenerated := job.Labels[ProwJobLabelGenerated]; !isGenerated || label != staleLabel {
				tempJobs = append(tempJobs, job)
			}
		}
		jobConfig.Presubmits[repo] = tempJobs
	}
	
	for repo, jobs := range jobConfig.Postsubmits {
		tempJobs := []prowconfig.Postsubmit{}
		for _, job := range jobs {
			if label, isGenerated := job.Labels[ProwJobLabelGenerated]; !isGenerated || label != staleLabel {
				tempJobs = append(tempJobs, job)
			}
		}
		jobConfig.Postsubmits[repo] = tempJobs
	}
}
```

Reply:
Yeah, what you suggest is what I would do normally, but I'm doing some Go reading and this is supposed to be more efficient (less allocations) and quite idiomatic Go way to filter out elements out of the slice...

@stevekuznetsov WDYT?

Reply:
Yeah I think the re-slicing (as originally implemented) is fine, pretty common in Go

Reply:
I'll keep it then.